### PR TITLE
White labeling support and other changes (ticket #28)

### PR DESCRIPTION
**Pull request by vsadda**

### CODE REVIEW

**Comment by vsadda** [comment]
This change set brings in complete white labeling support along with more advanced process models

File: minuet/symphony/Paragon/Program.cs [hunk 1]
```diff
@@ -111,27 +114,23 @@ public static void Main()
...
-                Logger.Error("exception source:" + ex.Source);
-                Logger.Error("exception targetSite:" + ex.TargetSite);
-                Logger.Error("exception type:" + ex.GetType());
-                Logger.Error("exception toString:" + ex.ToString());
-                ex = ex.InnerException;
+                var apps = _appManager.AllApplicaions.Select(app => app.Name).ToArray();
+                if (apps.Length > 0)
+                {
+                    errMessage += "\n\nThe following applications will be closed: \n    " + string.Join(", ", apps);
+                }
```
Comment (lneir) [inline minuet/symphony/Paragon/Program.cs:129]: you're loosing a lot of valuable debugging info here.  why remove code on lines 120 to 131?

Comment (vsadda) [inline minuet/symphony/Paragon/Program.cs:129]: The details are logged. It is not of any use to show these details to end users in an error message.

Comment (lneir) [inline minuet/symphony/Paragon/Program.cs:129]: these are added to the logger though and will be useful when gathering debug logs from the user.  they were already useful when we had a crash and the previous crash details did not provide inner exception.

Comment (vsadda) [inline minuet/symphony/Paragon/Program.cs:129]: The following line log the unhandled exception to the log file:

```
        logger.Error(string.Format("An unhandled domain exception has occurred: {0}", e.ExceptionObject ?? string.Empty));
```

What else do you think is being lost?

Comment (lneir) [inline minuet/symphony/Paragon/Program.cs:129]: the code above is digging into the innerException using the while loop and keeps digging down.  i do not believe the old code would provide this behavior.

Comment (vsadda) [inline minuet/symphony/Paragon/Program.cs:129]: I don't know why you believe so, but that is not correct. Formatting an exception object would result in a call to the ToString() method on that object, which would print the entire exception chain. If you want to you can try this following code.

```
        try
        {
            try
            {
                throw new Exception("Inner");
            }
            catch (Exception ex)
            {
                throw new Exception("Outer", ex);
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine("Error {0}", ex);
        }
```

Comment (lneir) [inline minuet/symphony/Paragon/Program.cs:129]: i did some testing and the new code seems to work fine.  it is indeed logging the full exception trace to the log file.

the exception where this wasn't working previously was for PR: https://github.com/SymphonyOSF/SFE-Minuet-DesktopClient/commit/07041d41956d5de191874f21079767e8b276a9da

**Pull request merged**